src/Native/System.Net.Http.Native/pal_easy.cpp

```diff
@@ -256,6 +267,31 @@ extern "C" int32_t HttpNative_RegisterSslCtxCallback(CURL* curl,
     return curl_easy_setopt(curl, CURLOPT_SSL_CTX_FUNCTION, &ssl_ctx_callback);
 }
 
+static int debug_callback(CURL* curl, curl_infotype type, char *data, size_t size, void *userPointer)
+{
+    assert(userPointer != nullptr);
+    CallbackHandle* handle = static_cast<CallbackHandle*>(userPointer);
+    handle->debugCallback(curl, static_cast<PAL_CurlInfoType>(type), data, size, handle->debugUserPointer);
+    return 0;
+}
+
+extern "C" int32_t HttpNative_RegisterDebugCallback(CURL* curl, 
+                                                    DebugCallback callback, 
+                                                    void* userPointer, 
+                                                    CallbackHandle** callbackHandle)
+{
+    EnsureCallbackHandle(callbackHandle);
+
+    CallbackHandle* handle = *callbackHandle;
+    handle->debugCallback = callback;
+    handle->debugUserPointer = userPointer;
+
+    CURLcode rv = curl_easy_setopt(curl, CURLOPT_DEBUGDATA, handle);
+    return rv == CURLE_OK ? 
+        curl_easy_setopt(curl, CURLOPT_DEBUGFUNCTION, &debug_callback) : 
+        rv;
```

Comment:
Nit: In case of a failure here, do we want to call the debugHandle to say that we cannot collect the traces?